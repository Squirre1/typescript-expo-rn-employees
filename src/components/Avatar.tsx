import React from 'react'
import { Avatar } from 'react-native-elements'

const CustomAvatar = (props: any) => props.uri ? (
  <Avatar
    xlarge={true}
    rounded={true}
    source={{ uri: props.uri }}
    activeOpacity={0.7}
    {...props}
  />
) : (
  <Avatar
    xlarge={true}
    rounded={true}
    icon={{name: 'ios-person', type: 'ionicon'}}
    activeOpacity={0.7}
    {...props}
  />
)

export default CustomAvatar
