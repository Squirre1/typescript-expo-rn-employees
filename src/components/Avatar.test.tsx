import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { Text } from 'react-native'
import { mount } from 'enzyme'
import * as chai from 'chai'

import Avatar from './Avatar'

it('renders without crashing', () => {
  const rendered = renderer.create(<Avatar uri="https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"/>).toJSON()
  expect(rendered).toMatchSnapshot()
  expect(rendered).toBeTruthy()
})

it('renders letters if uri not defined', () => {
  const wrapper = mount(<Avatar title="MM" />);
  chai.expect(wrapper.find(Text)).to.have.length(1);
})
