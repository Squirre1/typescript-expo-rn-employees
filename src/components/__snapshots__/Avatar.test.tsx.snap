// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  activeOpacity={0.7}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "height": 150,
        "paddingBottom": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": 150,
      },
      Object {
        "borderRadius": 75,
      },
      undefined,
    ]
  }
  uri="https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "rgba(0,0,0,0.2)",
          "bottom": 0,
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "borderRadius": 75,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
        }
      }
      style={
        Array [
          Object {
            "height": 150,
            "width": 150,
          },
          Object {
            "borderRadius": 75,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
